### PR TITLE
Added functions to get subnet mask given prefix length in CIDR notation, get broadcast and network addresses given ip interface and prefix length in CIDR notation

### PR DESCRIPTION
**Pull request by anchat1990**
These functionalities can currently handle/return IPv4 address format only

### CODE REVIEW

**Comment by anchat1990** [comment]
not sure why the checks are failing `npm test`?
My node version is v6.10.2

File: ipaddr.min.js [hunk 1]
```diff
@@ -1 +1 @@
...
\ No newline at end of file
+lib,ipaddr.js;
```
Comment (whitequark) [inline ipaddr.min.js]: You've removed the minified code completely.

File: src/ipaddr.coffee [hunk 2]
```diff
@@ -39,6 +52,41 @@ ipaddr.subnetMatch = (address, rangeList, defaultName='unicast') ->
 
   return defaultName
 
+# A utility function to return broadcast address given the IP interface and prefix length in CIDR notation
+ipaddr.broadcastAddressFromCIDR = (string) ->
+  try
+    ipInterface = ipaddr.IPv4.parseCIDR(string)[0]
+    subnetMask = @subnetMaskFromPrefixLength([ ipaddr.IPv4.parseCIDR(string)[1] ])
+    i = undefined
```
Comment (whitequark) [inline src/ipaddr.coffee]: This line is useless

File: src/ipaddr.coffee [hunk 3]
```diff
@@ -39,6 +52,41 @@ ipaddr.subnetMatch = (address, rangeList, defaultName='unicast') ->
...
+    octets = []
+    i = 0
+    while i < 4
+      # Broadcast address is bitwise OR between ip interface and inverted mask
+      octets.push parseInt(ipInterface.octets[i], 10) | parseInt(subnetMask.octets[i], 10) ^ 255
+      i++
+    return new (ipaddr.IPv4)(octets)
+  catch error
+    throw new Error('ipaddr: the address has neither IPv6 nor IPv4 CIDR format')
```
Comment (whitequark) [inline src/ipaddr.coffee]: I don't think this function actually handles IPv6

File: src/ipaddr.coffee [hunk 4]
```diff
@@ -39,6 +52,41 @@ ipaddr.subnetMatch = (address, rangeList, defaultName='unicast') ->
...
+    i = undefined
+    octets = []
+    i = 0
+    while i < 4
+      # Network address is bitwise AND between ip interface and mask
+      octets.push parseInt(ipInterface.octets[i], 10) & parseInt(subnetMask.octets[i], 10)
+      i++
+    return new (ipaddr.IPv4)(octets)
+  catch error
+    throw new Error('ipaddr: the address has neither IPv6 nor IPv4 CIDR format')
```
Comment (whitequark) [inline src/ipaddr.coffee]: I don't think this function actually handles IPv6

Comment (anchat1990) [inline src/ipaddr.coffee]: updated the error message.

File: src/ipaddr.coffee (hunk 3 above)
Comment (anchat1990) [inline src/ipaddr.coffee]: updated the error message.

File: src/ipaddr.coffee (hunk 2 above)
Comment (anchat1990) [inline src/ipaddr.coffee]: Apologies. I used js2coffee clearly has some hitches. Updated.

File: ipaddr.min.js (hunk 1 above)
Comment (anchat1990) [inline ipaddr.min.js]: Woops - updated and added back.

**Comment by whitequark** [comment]
Thanks. Your code looks great. There is just one more issue. The functions in the `ipaddr` namespace generally work on both IPv4 and IPv6 addresses. I think that either of these should be done:

  1. The new functions extended to handle IPv6 CIDR.
  2. The new functions moved under ipaddr.IPv4 namespace.

**Comment by anchat1990** [comment]
Thanks - I went with option 2. I've pushed the updated code.

File: README.md [hunk 5]
```diff
@@ -174,6 +174,22 @@ ipaddr.IPv4.parse('255.255.255.240').prefixLengthFromSubnetMask() == 28
 ipaddr.IPv4.parse('255.192.164.0').prefixLengthFromSubnetMask()  == null
 ```
 
+`subnetMaskFromPrefixLength()` will return an IPv4 netmask for a valid CIDR prefix length.
+
+```js
+ipaddr.subnetMaskFromPrefixLength("24") == "255.255.255.0"
```
Comment (whitequark) [inline README.md]: Outdated doc

Comment (anchat1990) [inline README.md]: Updated.

**Comment by whitequark** [comment]
OK. At last, please fix CI. I just built master on CI again and it still works so it is something that your change introduces.

**Pull request merged**